backend/service-user/src/main/java/g55/cs3219/backend/userService/service/AuthenticationService.java

```diff
@@ -30,6 +30,16 @@ public AuthenticationService(UserRepository userRepository, PasswordEncoder pass
     }
 
     public User signup(RegisterUserDto input) {
+        // Check for duplicate username
+        if (userRepository.findByUsername(input.getUsername()).isPresent()) {
+            throw new RuntimeException("Username already exists");
```

Comment:
Perhaps can create a new `Exception` class for this (a general version), i.e. `UserExistException`